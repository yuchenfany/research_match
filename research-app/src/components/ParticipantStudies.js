/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState, useEffect } from 'react';
import '../assets/index.css';
import { useNavigate } from 'react-router-dom';
import NavBar from './NavBar';

import { getEnrolledStudyIds } from '../modules/user-api';
import { getStudyById } from '../modules/study-api';

function ParticipantStudies({
  user, setUser, setStudy, setStatus,
}) { // add props user
  const [enrolledStudies, setEnrolledStudies] = useState([]);
  const navigate = useNavigate();

  // get all studies
  async function getAllStudyJson() {
    const studyIds = await getEnrolledStudyIds(user, setUser);
    return Promise.all(studyIds.map((studyId) => getStudyById(studyId)));
  }

  useEffect(() => {
    getAllStudyJson()
      .then(setEnrolledStudies);
  }, []);

  function goToStudy(studyId) {
    setStudy({ studyId });
    if (user.enrolled.indexOf(studyId) > -1) {
      setStatus({ isEnrolled: true });
    } else {
      setStatus({ isEnrolled: false });
    }
    navigate(`/study/${studyId}`);
  }

  return (
    <div className="Home">
      <NavBar user={user} />
      <div className="study-flex">
        <div className="header-left">Enrolled Studies</div>
        <div>
          {
          enrolledStudies.length === 0 ? []
            : enrolledStudies.map(
              (studyJson) => (
                <div key={studyJson.studyId} className="study">
                  <div className="study-title">{studyJson.title}</div>
                  <button className="view-button" type="button" key={studyJson.studyId} onClick={() => goToStudy(studyJson.studyId)}>VIEW</button>
                </div>
              ),
            )
          }
        </div>
      </div>
    </div>
  );
}

export default ParticipantStudies;
