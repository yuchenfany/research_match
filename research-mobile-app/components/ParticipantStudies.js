/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState, useEffect } from 'react';
import { View, Button, Text, StyleSheet } from 'react-native';

function ParticipantStudies({ route, navigation }) { // add props user
  const [enrolledStudies, setEnrolledStudies] = useState([]);
  const { user, setUser, setStudy, setStatus } = route.params;

  // gets list of all enrolled studies for user
  // NOTE: remove hardcoding once users are enrolled in studies

  async function getStudyIds() {
    const data = await fetch(`http://localhost:5000/record/${user.username}`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    const json = await data.json();
	user.enrolled = json.enrolled;
    setUser(user);
    return json?.enrolled ?? [];
  }

  // gets inViewidual study by id
  async function getStudy(studyId) {
    const data = await fetch(`http://localhost:5000/study/${studyId}`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    return data.json();
  }

  // get all studies
  async function getAllStudyJson() {
    const studyIds = await getStudyIds();
    return Promise.all(studyIds.map((studyId) => getStudy(studyId)));
  }

  useEffect(() => {
    getAllStudyJson()
      .then(setEnrolledStudies);
  }, []);

  function goToStudy(studyId) {
    setStudy({ studyId });
    if (user.enrolled.indexOf(studyId) > -1) {
      setStatus({ isEnrolled: true });
    } else {
      setStatus({ isEnrolled: false });
    }
	navigation.navigate('Study', route.params);
  }

  return (
    <View className="Home">
      <View className="study-flex">
        <Text className="header-left">Enrolled Studies</Text>
        <View>
          {
          enrolledStudies.length === 0 ? []
            : enrolledStudies.map(
              (studyJson) => (
                <View key={studyJson.studyId} className="study">
                  <Text className="study-title">{studyJson.title}</Text>
                  <Button
				  	className="view-button"
					title="VIEW"
					key={studyJson.studyId}
					onClick={() => goToStudy(2)}
				  >
				  </Button>
                </View>
              ),
            )
          }
        </View>
      </View>

      <View className="study-flex">
        <Text className="header-left">For Testing Purposes: Delete Later</Text>
        <View className="study">
          <Text className="study-title">Sleep Research</Text>
          <Button className="view-button" title="VIEW" key={2} onClick={() => goToStudy(2)}></Button>
        </View>
      </View>
      <View className="study-transfer">
        <Text className="header-left">For Testing Purposes: Directs to Add Study Page</Text>
        <View className="study">
          <Text className="study-transfer">Go to Study Page</Text>
          <Button
		  	className="view-button"
			title="Add Study"
			onClick={() => navigation.push('AddStudy', route.params)}
			>
		  </Button>
        </View>
      </View>
    </View>
  );
}

export default ParticipantStudies;
