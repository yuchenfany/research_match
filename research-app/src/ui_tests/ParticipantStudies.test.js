/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import ParticipantStudies from '../components/ParticipantStudies';

// const mockedUsedNavigate = jest.fn();

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useNavigate: () => mockedUsedNavigate,
// }));

test('Title For Testing Purposes: Delete Later', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantStudies  user={user1} setUser={mockSetUser} setStudy={mockSetStudy} setStatus={mockSetStatus} />
        </Router>);  
  const label = screen.getByText("For Testing Purposes: Delete Later");
  expect(label).toBeInTheDocument();
});

// test('participant clicked',() => {
//   render(
//     <Router>
//         <Study />
//     </Router>); 
// // fireEvent.click(screen.getByTestId('participant'));
// expect(screen.getByTestId('participant')).toBeInTheDocument();
// //   expect(screen.getByTestId('participant')).toBeChecked();
// });

test('View Button',() => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantStudies  user={user1} setUser={mockSetUser} setStudy={mockSetStudy} setStatus={mockSetStatus} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /view/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});

// // test('Researcher Button',() => {
// //   const mockSetUser = jest.fn();
// //   const user1 = { username: 'user1', password: 'user1pass' };
// //   render(
// //     <Router>
// //       <Type setUser={mockSetUser} user={user1} />
// //     </Router>,
// //   );
// //   const linkElement = screen.getByRole('radio', {
// //     name: /researcher/i
// //   });
// //   fireEvent.click(linkElement);
// //   // expect(mockSetUser).toBeCalled();
// //   expect(linkElement).toBeInTheDocument();
// // });
// test('researcher clicked',() => {
//   render(
//     <Router>
//         <Study />
//     </Router>); 
//   // fireEvent.click(screen.getByTestId('researcher'));
//   expect(screen.getByTestId('researcher')).toBeInTheDocument();
//   // expect(screen.getByTestId('researcher')).toBeChecked();
// });

test('snapshot test', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
  const component = renderer.create( 
    <Router>
        <ParticipantStudies  user={user1} setUser={mockSetUser} setStudy={mockSetStudy} setStatus={mockSetStatus} />
    </Router>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


//TODO: 
//1. Data not being accessed from queries