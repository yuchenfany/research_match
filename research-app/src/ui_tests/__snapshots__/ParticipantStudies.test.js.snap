// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Home"
>
  <div
    className="nav"
  >
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      HOME
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      MY STUDIES
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      MESSAGES
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      ANALYTICS
    </button>
    <button
      className="account-btn"
      onClick={[Function]}
      type="button"
    >
      ACCOUNT
    </button>
    <div
      className="acct-options"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <button
        className="account-btn"
        onClick={[Function]}
        type="button"
      >
        EDIT PROFILE
      </button>
      <button
        className="account-btn"
        onClick={[Function]}
        type="button"
      >
        DELETE ACCOUNT
      </button>
      <button
        className="logout-btn"
        onClick={[Function]}
        type="button"
      >
        LOGOUT
      </button>
    </div>
  </div>
  <div
    className="study-flex"
  >
    <div
      className="header-left"
    >
      Enrolled Studies
    </div>
    <div />
  </div>
  <div
    className="study-flex"
  >
    <div
      className="header-left"
    >
      For Testing Purposes: Delete Later
    </div>
    <div
      className="study"
    >
      <div
        className="study-title"
      >
        Sleep Research
      </div>
      <button
        className="view-button"
        onClick={[Function]}
        type="button"
      >
        VIEW
      </button>
    </div>
  </div>
</div>
`;
